docs/rationale.md

```diff
@@ -0,0 +1,180 @@
+sequence of properties that can be used from the global scope. For example,
+tests for [the `Array.prototype.reduce`
+method](https://tc39.es/ecma262/#sec-array.prototype.reduce) are located in
+[the `tests/built-ins/Array/prototype/reduce`
+directory](https://github.com/tc39/test262/tree/main/test/built-ins/Array/prototype/reduce),
+while tests for [the `isNaN`
+function](https://tc39.es/ecma262/#sec-isnan-number) are located in [the
+`tests/built-ins/isNan`
+directory(https://github.com/tc39/test262/tree/main/test/built-ins/isNaN)].
+Built-ins which are defined only in [the ECMA-402
+specification](https://tc39.es/ecma402/) follow a similar naming convention
+within [the `tests/intl402`
+directory](https://github.com/tc39/test262/tree/main/test/intl402).
+
+[The `tests/annexB`
+directory](https://github.com/tc39/test262/tree/main/test/annexB) holds tests
+for the semantics described by [Annex B of
+ECMA262](https://tc39.es/ecma262/#sec-additional-ecmascript-features-for-web-browsers).
+The conventions for syntax-derived operations and built-in APIs as described
+above are also applied within this directory.
+
+[The `tests/harness`
+directory](https://github.com/tc39/test262/tree/main/test/harness) stores tests
+for the "harness" files which Test262 maintains to assist in test writing.
+
+Directories are not generally applied beyond these limits; further
+differentiation is instead achieved through structured file names which follow
+ad-hoc conventions. This organization balances the need to group tests
+logically with the need to discover tests. See, for example, [the tests for
+template
+literals](https://github.com/tc39/test262/tree/main/test/language/expressions/template-literal).
+
+Many consumers use file names as a way to compare test results across revisions
+and between implementations. For this reason, tests files are rarely
+re-organized after being accepted.
+
+## Regression tests
+
+It is possible to write tests for semantics which, while not explicitly
+specified by ECMA262, are nonetheless valid according to the normative text.
+Such tests are welcome in Test262, but their fitness is not a given. Consumers
+from many constituencies value the coherence and consistency of the test suite,
+and tests which disallow arbitrary extraneous behavior can degrade those
+qualities. Because Test262 is not maintained as a repository of regression
+tests, contributions which include these kinds of tests will be weighed against
+their likelihood of identifying error in a plurality of implementations.
+
+For example, assume that some runtime spuriously accesses the `toJSON` property
+of the value passed to
+[`String.prototype.repeat`](https://tc39.es/ecma262/#sec-string.prototype.repeat).
+While the maintainers of the engine may decide to include a regression test
+which disallows that behavior in their project, the maintainers of Test262
+would not necessarily accept such a test.
```

Comment:
i'm not sure this matches the first paragraph in the section - doing anything observable that's not in the spec for a method is definitely prohibited.

Reply:
I think Mike's point is that accessing `.toJSON` as part of `String.prototype.repeat`, while prohibited, is a sufficiently unlikely and out-there spec violation that it's not a good use of test262 bandwidth to encourage people to write tests for it.

Reply:
I agree - right up until an implementation does it. That makes it infinitely likely (since it actually happened) and at that point imo we should be strongly inclined to test against it.

Reply:
@ptomato's interpretation is pretty close to my intent, though it maybe under-emphasizes the part about "identifying error in a plurality of implementations." In any case, I'm confident this has been watered down enough already--the phrase "would not necessarily accept such a test" leaves plenty of room for interpretation.